// ng2-adsense/esm2015/adsense.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ADSENSE_TOKEN } from './adsense-config';
import { AdsenseComponent } from './adsense.component';
export class AdsenseModule {
    /**
     * @param {?=} config
     * @return {?}
     */
    static forRoot(config = {}) {
        return {
            ngModule: AdsenseModule,
            providers: [{ provide: ADSENSE_TOKEN, useValue: config }],
        };
    }
}
AdsenseModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                exports: [AdsenseComponent],
                declarations: [AdsenseComponent],
            },] }
];
function AdsenseModule_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    AdsenseModule.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    AdsenseModule.ctorParameters;
}
